app/bundles/EmailBundle/Controller/PublicController.php

```diff
@@ -107,6 +107,25 @@ public function trackingImageAction(
         return TrackingPixelHelper::getResponse($request);
     }
 
+    /**
+     * One-Click unsubscribe - RFC 8058.
+     */
+    public function oneClickUnsubscribeAction(EmailModel $emailModel, string $idHash): Response
+    {
+        $stat = $emailModel->getEmailStatus($idHash);
+        if (!empty($stat)) {
+            $unsubscribeComment = $this->translator->trans('mautic.email.dnc.unsubscribed');
+            $emailModel->setDoNotContact($stat, $unsubscribeComment, DoNotContact::UNSUBSCRIBED);
+        } else {
+            $content = $this->translator->trans('mautic.email.stat_record.not_found');
+
+            return new Response($content, Response::HTTP_NOT_FOUND);
+        }
+        $content = $this->translator->trans('mautic.lead.do.not.contact_unsubscribed');
+
+        return new Response($content);
```

Comment:
We could avoid the `else` part and make it less complex
```suggestion
        $stat = $emailModel->getEmailStatus($idHash);

        if (!$stat instanceof Stat) {
            return new Response($this->translator->trans('mautic.email.stat_record.not_found'), Response::HTTP_NOT_FOUND);
        }

        $unsubscribeComment = $this->translator->trans('mautic.email.dnc.unsubscribed');
        $emailModel->setDoNotContact($stat, $unsubscribeComment, DoNotContact::UNSUBSCRIBED);

        return new Response($this->translator->trans('mautic.lead.do.not.contact_unsubscribed'));
```

Reply:
I did my best to keep it simple while moving to the existing method